src/api-service/__app__/onefuzzlib/workers/scalesets.py

```diff
@@ -551,6 +545,9 @@ def delete_nodes(self, nodes: List[Node]) -> None:
             machine_ids,
         )
         delete_vmss_nodes(self.scaleset_id, machine_ids)
+        for node in nodes:
+            if node.machine_id in machine_ids:
+                node.delete()
```

Comment:
Can we just iterate over `machine_ids` directly? By construction, its IDs are a subset of those in `nodes`, right? If there's a reason to iterate over `nodes` instead, can you document it?

Reply:
No?  In order to delete the `Node` object, we need the `Node` instance.

`nodes` is of type `Node`.  machine_ids is `List[UUID]`.

We could iterate the other way around, but it would be less efficient.

Reply:
Ah, right right. Mental typechecker failure.